test/test_coap_listen.py

```diff
@@ -164,26 +117,29 @@ def test_plugin_reconfigure_else(mocker, unused_port):
     mocker.patch.dict(config, {'port': port})
     config['port']['value'] = config['port']['default']
     config['uri']['value'] = config['uri']['default']
-    log_exception = mocker.patch.object(coap._LOGGER, "exception")
+    new_config = copy.deepcopy(_NEW_CONFIG)
+    new_config['port']['value'] = new_config['port']['default']
+    new_config['uri']['value'] = new_config['uri']['default']
+    pstop = mocker.patch.object(coap, 'plugin_shutdown', return_value=True)
     log_info = mocker.patch.object(coap._LOGGER, "info")
 
     # WHEN
-    loop.call_soon(coap.plugin_start(config))
-    await asyncio.sleep(.5)  # required to allow ensure_future task to complete
-    coap._plugin_stop(config)
+    new_handle = coap.plugin_reconfigure(config, new_config)
+    await asyncio.sleep(.3)
```

Comment:
why this sleep is required. last sleep statement had a comment next to that.

Reply:
done